src/include/lxcmntent.c

```diff
@@ -154,20 +154,24 @@ struct mntent *getmntent(FILE *stream)
 /* Prepare to begin reading and/or writing mount table entries from the
  * beginning of FILE. MODE is as for `fopen'.
  */
+#define __SETMNTENT_MODE_MAX 256
 FILE *setmntent(const char *file, const char *mode)
 {
 	/* Extend the mode parameter with "c" to disable cancellation in the
 	 * I/O functions and "e" to set FD_CLOEXEC.
 	 */
 	size_t modelen = strlen(mode);
-	char *newmode;
+	char newmode[__SETMNTENT_MODE_MAX];
 
-	newmode = alloca(modelen + 3);
+	if (modelen >= (__SETMNTENT_MODE_MAX - 2)) {
```

Comment:
The `strlen()` doesn't include the `\0` so this should be `3` - equivalent to the `"ce\0"` we append.

Reply:
Thanks for spotting. Can you go through all of it now, please? 😬